src/frameworks/fastify.js

```diff
@@ -0,0 +1,44 @@
+function validate(validateRequest) {
+    return (request, reply) => {
+        const requestOptions = _getParameters(request);
+        return validateRequest(requestOptions).then(function (errors) {
+            if (errors) {
+                throw errors;
+            }
+        });
+    };
+}
+
+function _getParameters(req) {
+    const requestOptions = {};
+    const path = req.raw.url;
+    requestOptions.path = path.endsWith('/') ? path.substring(0, path.length - 1) : path;
+    requestOptions.headers = req.headers;
+    requestOptions.params = req.params;
+    requestOptions.query = req.query;
+    requestOptions.files = req.files;
+    requestOptions.method = req.raw.method;
+    requestOptions.body = req.body;
+
+    return requestOptions;
+}
+
+function fastifyValidationPlugin(validateRequestFn) {
+    const fp = require('fastify-plugin');
+
+    return fp(function (fastify, options, next) {
+        fastify.addHook('onRequest', validate(validateRequestFn));
+        next();
+    });
+}
+
+function getPlugins(validateRequestFn) {
```

Comment:
is there a reason for not exposing the plugin as koa/express middleware are exposed and not through a dedicated function for getting the plugin?

Reply:
Yes. If you take a look at `fastifyValidationPlugin`, it needs a `validateRequestFn` as a parameter, which is not available until .init() was invoked on express-ajv-swagger-validation. Hence we need to defer instantiation of the plugin until `init` was invoked, and there is no existing mechanism for that.

Reply:
However, I see there is a gap that we instantiate plugins on every call, which shouldn't be the case. let me fix it quickly

Reply:
Addressed the gap, only init once.